Fleece/Core/Doc.cc

```diff
@@ -306,6 +306,20 @@ namespace fleece { namespace impl {
         return RetainedConst<Doc>((const Doc*)scope);
     }
 
+
+    bool Doc::setAssociated(void *pointer, const char *type) {
+        if (_associatedType && type && (_associatedType != type))
+            return false;
+        _associatedPointer = pointer;
+        _associatedType = type;
+        return true;
+    }
+
+    void* Doc::getAssociated(const char *type) const {
+        return (type == _associatedType || type == nullptr) ? _associatedPointer : nullptr;
```

Comment:
If type is nullptr but _associatedType is not nullptr, the _associatedPointer is also returned. Is this expected?

Reply:
Yeah, I left it as a way to get the pointer no matter what type it is. Not encouraged, but it could be useful for debugging or logging.